src/controllers/expense.controller.js

```diff
@@ -0,0 +1,150 @@
+/* eslint-disable no-console */
+'use strict';
+
+const {
+  STATUS_CODE_OK,
+  STATUS_CODE_CREATED,
+  STATUS_CODE_NO_CONTENT,
+  STATUS_CODE_BAD_REQUEST,
+  STATUS_CODE_NOT_FOUND,
+} = require('../constants');
+
+const expenseService = require('../services/expense.service');
+const userService = require('../services/user.service');
+
+const getAll = (req, res) => {
+  const {
+    userId,
+    categories,
+    from,
+    to,
+  } = req.query;
+
+  let filteredExpenses = expenseService.getAll();
+
+  if (userId) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => expense.userId === +userId);
+  }
+
+  if (categories) {
+    filteredExpenses = filteredExpenses.filter(
+      ({ category }) => categories.includes(category)
```

Comment:
if there are no `categories` in search params we will get `undefined`
if only 1 `categories` we will get just value (....?categories=1)
And if we have many categories we will get array of values (....?categories=1&categories=2&categories=3)

![image](https://github.com/mate-academy/node_accounting-app-v2/assets/86789161/ea010564-92a2-4116-9fcf-a66739d6d01f)

Your code will crash if you have only 1 category